src/wallet/rpcwallet.cpp

```diff
@@ -3102,12 +3114,16 @@ UniValue loadwallet(const JSONRPCRequest& request)
 
 UniValue createwallet(const JSONRPCRequest& request)
 {
-    if (request.fHelp || request.params.size() != 1) {
+    if (request.fHelp || request.params.size() < 1 || request.params.size() > 2) {
         throw std::runtime_error(
-            "createwallet \"wallet_name\"\n"
+            "createwallet \"wallet_name\" (\"options\"\n"
             "\nCreates and loads a new wallet.\n"
             "\nArguments:\n"
             "1. \"wallet_name\"    (string, required) The name for the new wallet. If this is a path, the wallet will be created at the path location.\n"
+            "2. options          (json, optional)\n"
+            "  {\n"
+            "     \"disableprivatekeys\": <state>,         (boolean, optional, default: false) Disable the possibility of private keys (only watchonlys are possible in this mode).\n"
```

Comment:
I was under the impression that named arguments is preferable to `options` object.

Reply:
I agree. Named arguments are much preferable. Using named arguments provides type- and error- checking for free, and simplify/flatten the interface

For example, if I typo the an option `disableprivtaekeys` here, this implementation will silently ignore the option. That's not possible with named arguments.